### DISCUSSION TITLE
What does classesDir define now?

### DISCUSSION THREAD

**Discussion by davidmlw** [scalacenter/bloop#1256] [General] [answered]
From documents: https://scalacenter.github.io/bloop/docs/configuration-format, classesDir defines the output directory for the compiled binaries. While I saw the current bloop v1.4.0-RC1 does not respect this directory and put the files in .bloop/module/bloop-internal-classes/bloop-cli-RANDOM/.

**Comment by davidmlw** [comment]
And the directory  `.bloop/module/bloop-internal-classes/bloop-cli-RANDOM/` is not only random named, but is also random generated with multiple versions. how to get a stable classesDir ?

**Comment by jvican** [comment, marked as answer]
Hi @davidmlw, have you had a look at https://github.com/scalacenter/bloop/issues/1197 ? It explains what the `classesDir` field in the configuration file is used for and for what use case.
  
> And the directory .bloop/module/bloop-internal-classes/bloop-cli-RANDOM/ is not only random named, but is also random generated with multiple versions. how to get a stable classesDir ?
  
The directory for the CLi is stable, but it can be random if you have several CLI commands running.

**Reply by davidmlw** [reply]
I see. My build flow tool emit multiple bloop CLI compile commands, so bloop make different directory. That is why I saw different random directory.

**Reply by jvican** [reply]
What was the reason you looked into where the classes directories were saved?
